src/api/checkout.js

```diff
@@ -4,6 +4,13 @@ import { addAction } from '../lib/database';
 let app = express();
 
 app.post('/', (req, res) => {
+    if (typeof req.body.studentId !== 'string') {
+        return res.failureJson('A student id is required.');
```

Comment:
Optional in the sense that nobody is forcing it. Required in the sense that it doesn't conform to the HTTP spec.